test/unittests/hex_test.cpp

```diff
@@ -86,34 +74,35 @@ TEST(hex, from_hex_skip_whitespace)
 
 TEST(hex, validate_hex)
 {
-    EXPECT_FALSE(validate_hex(""));
-    EXPECT_FALSE(validate_hex("0x"));
-    EXPECT_FALSE(validate_hex("01"));
-    EXPECT_EQ(validate_hex("0"), hex_errc::incomplete_hex_byte_pair);
-    EXPECT_EQ(validate_hex("WXYZ"), hex_errc::invalid_hex_digit);
-}
-
-TEST(hex, hex_error_code)
-{
-    std::error_code ec1 = hex_errc::invalid_hex_digit;
-    EXPECT_EQ(ec1.value(), 1);
-    EXPECT_EQ(ec1.message(), "invalid hex digit");
-
-    std::error_code ec2 = hex_errc::incomplete_hex_byte_pair;
-    EXPECT_EQ(ec2.value(), 2);
-    EXPECT_EQ(ec2.message(), "incomplete hex byte pair");
-}
-
-TEST(hex, hex_category_inspection)
-{
-    EXPECT_STREQ(hex_category().name(), "hex");
+    EXPECT_TRUE(validate_hex(""));
+    EXPECT_TRUE(validate_hex("0x"));
+    EXPECT_TRUE(validate_hex("01"));
+    EXPECT_FALSE(validate_hex("0"));
+    EXPECT_FALSE(validate_hex("WXYZ"));
 }
 
-TEST(hex, hex_category_comparison)
+TEST(hex, isspace)
 {
-    std::error_code ec1 = hex_errc::invalid_hex_digit;
-    EXPECT_EQ(ec1.category(), hex_category());
+    // Test internal isspace() compliance with std::isspace().
+    // The https://en.cppreference.com/w/cpp/string/byte/isspace has the list of "space" characters.
 
-    std::error_code ec2 = hex_errc::incomplete_hex_byte_pair;
-    EXPECT_EQ(ec2.category(), hex_category());
+    for (int i = int{std::numeric_limits<char>::min()}; i <= std::numeric_limits<char>::max(); ++i)
+    {
+        const auto c = static_cast<char>(i);
+        EXPECT_EQ(evmc::internal_hex::isspace(c), (std::isspace(c) != 0));
+        switch (c)
+        {
+        case ' ':
+        case '\f':
+        case '\n':
+        case '\r':
+        case '\t':
+        case '\v':
```

Comment:
I actually wonder if `isspace` should be used in the hex parser or just checking for space would be enough?

Reply:
I need at least ` `, `\r`, `\n`.